src/main/java/org/walkmod/WalkModFacade.java

```diff
@@ -219,6 +223,58 @@ public WalkModFacade() {
 		this(false, true, false);
 	}
 
+
+	/**
+	 * Initalizes a Walkmod service
+	 *
+	 * @param walkmodCfg
+	 *            Walkmod configuration file. If null, file named 'walkmod.xml'
+	 *            is searched at the root.
+	 * @param options
+	 *            Map of option. See {@link Options} and {@link OptionsBuilder}
+	 *            for available options and default values.
+	 * @param configurationProvider
+	 *            Configuration provider responsible for the resolution of
+	 *            plugins (used to use custom classloading strategies).
+	 *            If null Ivy is used.
+	 */
+	public WalkModFacade(File walkmodCfg, Map<String,Object> options, ConfigurationProvider configurationProvider) {

```

Comment:
Personally, I would prefer that argument types restrict more because is less prune error. However, I understand that it becomes easier to use with a Map with Groovy and we need less code for that. 

All has its benefits and drawbacks. So, ok,  don't change the code and let's see what happens in a future if developers use it correctly.

BTW: Thanks for all your work :+1: